modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptDocsGenerator.cs

```diff
@@ -0,0 +1,268 @@
+            }
+
+            if (string.IsNullOrWhiteSpace(classDescription) && docPropertyString.Length == 0 && docSignalString.Length == 0)
+            {
+                // Script has no docs.
+                return;
+            }
+
+            INamespaceSymbol namespaceSymbol = symbol.ContainingNamespace;
+            string classNs = namespaceSymbol is { IsGlobalNamespace: false }
+                ? namespaceSymbol.FullQualifiedNameOmitGlobal()
+                : string.Empty;
+            bool hasNamespace = classNs.Length != 0;
+
+            bool isInnerClass = symbol.ContainingType != null;
+
+            string uniqueHint = symbol.FullQualifiedNameOmitGlobal().SanitizeQualifiedNameForUniqueHint()
+                                + "_ScriptDocs.generated";
+
+            var source = new StringBuilder();
+
+            if (hasNamespace)
+            {
+                source.Append("namespace ");
+                source.Append(classNs);
+                source.Append(" {\n\n");
+            }
+
+            if (isInnerClass)
+            {
+                var containingType = symbol.ContainingType;
+                AppendPartialContainingTypeDeclarations(containingType);
+
+                void AppendPartialContainingTypeDeclarations(INamedTypeSymbol? containingType)
+                {
+                    if (containingType == null)
+                        return;
+
+                    AppendPartialContainingTypeDeclarations(containingType.ContainingType);
+
+                    source.Append("partial ");
+                    source.Append(containingType.GetDeclarationKeyword());
+                    source.Append(" ");
+                    source.Append(containingType.NameWithTypeParameters());
+                    source.Append("\n{\n");
+                }
+            }
+
+            source.Append("partial class ");
+            source.Append(symbol.NameWithTypeParameters());
+            source.Append("\n{\n");
+
+            source.Append("#pragma warning disable CS0109 // Disable warning about redundant 'new' keyword\n");
+            source.Append("#if TOOLS\n");
+
+            source.Append("    [global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]\n");
+            source.Append("    internal new static global::Godot.Collections.Dictionary GetGodotClassDocs()\n    {\n");
+            source.Append("        var docs = new global::Godot.Collections.Dictionary();\n");
+            source.Append("        docs.Add(\"name\",\"");
+            if (symbol.GetAttributes().Any(a => a.AttributeClass?.IsGodotGlobalClassAttribute() ?? false))
+            {
+                source.Append(symbol.Name);
+            }
+            else
+            {
+                source.Append("\\\"" + ScriptPathAttributeGenerator.RelativeToDir(symbol.DeclaringSyntaxReferences.First().SyntaxTree.FilePath, godotProjectDir) + (isInnerClass ? ":" + symbol.Name : "") + "\\\"");
```

Comment:
Looks like GDScript uses a dot (`.`) and the inner class comes after the quoted path (not inside).

![image](https://github.com/user-attachments/assets/acf393a8-abe8-4b17-b33f-f980e7d9dae8)

```suggestion
                source.Append($"\\\"{ScriptPathAttributeGenerator.RelativeToDir(symbol.DeclaringSyntaxReferences.First().SyntaxTree.FilePath, godotProjectDir)}\\\"");
                if (isInnerClass)
                {
                    source.Append($".{symbol.Name}");
                }
```